app/routes/surveys/components/SurveyEditor/SurveyForm.tsx

```diff
@@ -0,0 +1,236 @@
+import { SubmitButton } from 'app/components/Form/SubmitButton';
+import Time from 'app/components/Time';
+import { EVENT_CONSTANTS, eventTypeToString } from 'app/routes/events/utils';
+import Question from 'app/routes/surveys/components/SurveyEditor/Question';
+import {
+  hasOptions,
+  initialQuestion,
+} from 'app/routes/surveys/components/SurveyEditor/utils';
+import styles from 'app/routes/surveys/components/surveys.css';
+import { spyValues } from 'app/utils/formSpyUtils';
+import { createValidator, required } from 'app/utils/validation';
+import type { EventType } from 'app/store/models/Event';
+import type { FormSurvey, FormSubmitSurvey } from 'app/store/models/Survey';
+import type { FormSurveyQuestion } from 'app/store/models/SurveyQuestion';
+import type { FieldArrayRenderProps } from 'react-final-form-arrays';
+
+const TypedLegoForm = LegoFinalForm<FormSurvey>;
+
+const validate = createValidator(
+  {
+    title: [required()],
+    event: [required()],
+  },
+  hasOptions
+);
+
+type Props = {
+  isNew?: boolean;
+  onSubmit: (surveyData: FormSubmitSurvey) => Promise<void>;
+  initialValues: Partial<FormSurvey>;
+  templateType: EventType | undefined;
+  setTemplateType: (templateType: EventType) => void;
+};
+
+const SurveyForm = ({
+  isNew,
+  onSubmit,
+  initialValues,
+  templateType,
+  setTemplateType,
+}: Props) => {
+  const [templatePickerOpen, setTemplatePickerOpen] = useState(false);
+
+  const internalOnSubmit = (surveyData: FormSurvey) => {
+    return onSubmit({
+      ...surveyData,
+      event: surveyData.event.value,
+      questions: surveyData.questions.map((question) => ({
+        ...question,
+        questionType: question.questionType.value,
+        options: question.options.slice(0, -1),
+      })),
+    });
+  };
+
+  return (
+    <TypedLegoForm
+      onSubmit={internalOnSubmit}
+      validate={validate}
+      initialValues={initialValues}
+      subscription={{}}
+      mutators={{
+        ...arrayMutators,
+      }}
+    >
+      {({ handleSubmit }) => (
+        <form onSubmit={handleSubmit}>
+          <Field
+            placeholder="Tittel"
+            label=" "
+            name="title"
+            component={TextInput.Field}
+            className={styles.editTitle}
+          />
+          {templateType && (
+            <div className={styles.templateType}>
+              <span>
+                Bruker mal: <i>{EVENT_CONSTANTS[templateType]}</i>
+              </span>
+            </div>
+          )}
+          <div className={styles.templatePicker}>
+            <ConfirmModal
+              title="Bekreft bruk av mal"
+              message={
+                'Dette vil slette alle ulagrede endringer i undersøkelsen!\n' +
+                'Lagrede endringer vil ikke overskrives før du trykker "Lagre".'
+              }
+              closeOnConfirm
+              onCancel={async () => setTemplatePickerOpen(false)}
+              onConfirm={async () => setTemplatePickerOpen(true)}
```

Comment:
Is the `async` needed here?  You're not awaiting anything

Reply:
Not really, but the prop-types in `ConfirmModal` require it.